huaweicloud/services/cse/resource_huaweicloud_cse_microservice_engine.go

```diff
@@ -0,0 +1,316 @@
+	if len(creds) < 1 {
+		return nil
+	}
+
+	cred := creds[0].(map[string]interface{})
+
+	return &engines.AuthCred{
+		Password: cred["password"].(string),
+	}
+}
+
+func resourceMicroserviceEngineCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.CseV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSE v2 client: %s", err)
+	}
+
+	networkId := d.Get("network_id").(string)
+	subnetResp, err := vpc.GetVpcSubnetById(conf, region, networkId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	vpcResp, err := vpc.GetVpcById(conf, region, subnetResp.VPC_ID)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	authType := d.Get("auth_type").(string)
+	createOpts := engines.CreateOpts{
+		Payment:             "1",
+		SpecType:            "CSE2",
+		Name:                d.Get("name").(string),
+		Description:         d.Get("description").(string),
+		Flavor:              d.Get("flavor").(string),
+		AvailabilityZones:   utils.ExpandToStringList(d.Get("availability_zones").([]interface{})),
+		AuthType:            authType,
+		VpcName:             vpcResp.Name,
+		VpcId:               vpcResp.ID,
+		NetworkId:           networkId,
+		SubnetCidr:          subnetResp.CIDR,
+		PublicIpId:          d.Get("eip_id").(string),
+		Inputs:              d.Get("extend_params").(map[string]interface{}),
+		EnterpriseProjectId: common.GetEnterpriseProjectID(d, conf),
+	}
+
+	if authType == "RBAC" {
+		createOpts.AuthCred = buildAuthCredStructure(d.Get("auth_cred").([]interface{}))
+	}
+
+	resp, err := engines.Create(client, createOpts)
+	if err != nil {
+		return diag.Errorf("error creating Microservice engine: %s", err)
+	}
+	d.SetId(resp.ID)
+
+	log.Printf("[DEBUG] Waiting for the Microservice engine to become running, the engine ID is %s.", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Init", "Executing"},
+		Target:       []string{"Finished"},
+		Refresh:      MicroserviceJobRefreshFunc(client, d.Id(), strconv.Itoa(resp.JobId)),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        180 * time.Second,
+		PollInterval: 15 * time.Second,
+	}
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for the creation of Microservice engine (%s) to complete: %s", d.Id(), err)
+	}
+
+	return resourceMicroserviceEngineRead(ctx, d, meta)
+}
+
+func resourceMicroserviceEngineRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.CseV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating CSE v2 client: %s", err)
+	}
+
+	resp, err := engines.Get(client, d.Id(), common.GetEnterpriseProjectID(d, conf))
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Microservice engine")
```

Comment:
pls use parseEngineJobError to wrap the error

Reply:
done.